### PR TITLE
MRG: More readable __repr__s

### LINKED ISSUE #6096: Only 0 head digitization points of the specified kinds ("eeg", "extra",), at least 4 required
Dear MNE developers,
we are having trouble running maxfilter on an MEG data file recorded at NeuroSpin - 
apparently we are missing some head digitization points.
We did digitize extra points and the HPI fitting went fine, so I am not sure what went wrong.
Do you have any ideas on how to debug the problem and still make maxfilter run?
Thank you!
Sophie 

>   File "/home/sh252228/anaconda3/envs/mne/lib/python3.6/site-packages/mne/bem.py", line 931, in get_fitting_dig
>     raise ValueError(msg + ', at least 4 required')
> 
> ValueError: Only 0 head digitization points of the specified kinds ("eeg", "extra",), at least 4 required

Here is the code:
`  raw_sss = mne.preprocessing.maxwell_filter(
            raw,
            calibration=config.mf_cal_fname,
            cross_talk=config.mf_ctc_fname,
            st_duration=None,
            origin='auto',
            destination=destination)`

Destination is info['dev_head_t'] from the same run.

Here is the raw.info['dig']: 

> [{'kind': 1, 'ident': 1, 'r': array([-7.1110763e-02, -1.4901161e-08, -1.4901161e-08], dtype=float32), 'coord_frame': 4}, {'kind': 1, 'ident': 2, 'r': array([ 7.4505806e-09,  9.4172612e-02, -1.4901161e-08], dtype=float32), 'coord_frame': 4}, {'kind': 1, 'ident': 3, 'r': array([ 7.0182838e-02, -1.4901161e-08, -1.4901161e-08], dtype=float32), 'coord_frame': 4}, {'kind': 2, 'ident': 1, 'r': array([0.05354978, 0.08048022, 0.05662574], dtype=float32), 'coord_frame': 4}, {'kind': 2, 'ident': 2, 'r': array([ 0.06554938, -0.02938803, -0.01712441], dtype=float32), 'coord_frame': 4}, {'kind': 2, 'ident': 3, 'r': array([-0.06564789, -0.02865517, -0.00968074], dtype=float32), 'coord_frame': 4}, {'kind': 2, 'ident': 4, 'r': array([-0.05240473,  0.08593322,  0.04769342], dtype=float32), 'coord_frame': 4}]

### PR DESCRIPTION
**Pull request by larsoner**
This PR:

1. Updates `maxwell_filter` docstring to make it clearer what `origin='auto'` does and thus how to deal with errors.
2. Adds `DigPoint` class that subclasses `dict` to give a nice `__repr__` for each dig point (should not interfere with eventual dig refactoring, it's something we will want)
3. Updates `Info.__repr__` to actually tell you the `dig` content
4. Updates all `mne.io.constants.FIFF` entries to be magically more readable

# master
```
>>> import mne
>>> from pprint import pprint
>>> info = mne.io.read_info('s190301_TimeInWM_run01_ERROR_raw.fif')
>>> print(info)
<Info | 26 non-empty fields
    ...
    dig : list | 7 items
    ...
>
```
```
>>> pprint(info['dig'][:6])
[{'coord_frame': 4,
  'ident': 1,
  'kind': 1,
  'r': array([-7.1110763e-02, -1.4901161e-08, -1.4901161e-08], dtype=float32)},
 {'coord_frame': 4,
  'ident': 2,
  'kind': 1,
  'r': array([ 7.4505806e-09,  9.4172612e-02, -1.4901161e-08], dtype=float32)},
 {'coord_frame': 4,
  'ident': 3,
  'kind': 1,
  'r': array([ 7.0182838e-02, -1.4901161e-08, -1.4901161e-08], dtype=float32)},
 {'coord_frame': 4,
  'ident': 1,
  'kind': 2,
  'r': array([0.05354978, 0.08048022, 0.05662574], dtype=float32)},
 {'coord_frame': 4,
  'ident': 2,
  'kind': 2,
  'r': array([ 0.06554938, -0.02938803, -0.01712441], dtype=float32)},
 {'coord_frame': 4,
  'ident': 3,
  'kind': 2,
  'r': array([-0.06564789, -0.02865517, -0.00968074], dtype=float32)}]
```
```
>>> print(info['dig'][0]['coord_frame'])
4
```

# This PR
```
>>> print(info)
<Info | 26 non-empty fields
    ...
    dig : list | 7 items (3 Cardinal, 4 HPI)
    ...
>
```
```
>>> pprint(info['dig'][:6])
[<DigPoint |        LPA : (-71.1, -0.0, -0.0) mm    : head frame>,
 <DigPoint |     Nasion : (0.0, 94.2, -0.0) mm      : head frame>,
 <DigPoint |        RPA : (70.2, -0.0, -0.0) mm     : head frame>,
 <DigPoint |     HPI #1 : (53.5, 80.5, 56.6) mm     : head frame>,
 <DigPoint |     HPI #2 : (65.5, -29.4, -17.1) mm   : head frame>,
 <DigPoint |     HPI #3 : (-65.6, -28.7, -9.7) mm   : head frame>]
```
```
>>> print(info['dig'][0]['coord_frame'])
4 (FIFFV_COORD_HEAD)
```
That last one should really excite anyone who has worked with the `FIFF` constants. Countless times I've wanted to know what a given `int` from a `FIFF` entry corresponded to -- and now we get it without having to dig through code, thanks to the `BunchConstNamed` class.

Closes #6096.

### CODE REVIEW

**Comment by larsoner** [comment]
Pushed a commit to fix the Sphinx failure (they release 2.0.0 on PyPi 2 hours ago) that is presumably in master, it's a workaround until https://github.com/ryan-roemer/sphinx-bootstrap-theme/pull/193 lands

**Comment by larsoner** [comment]
@SophieHerbst would this docstring have helped:

https://12004-1301584-gh.circle-artifacts.com/0/dev/generated/mne.preprocessing.maxwell_filter.html

**Comment by SophieHerbst** [comment]
@ Larsoner
Docstring sounds great. Maybe a small note why the automatic fitting could fail? 
Along these lines: due to unsufficient number of digitization points / faulty HPI fit during recording?

**Comment by larsoner** [comment]
Tweaked to mention too few dig points

File: mne/utils/_bunch.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+"""Bunch-related classes."""
```
Comment (wmvanvliet) [inline mne/utils/_bunch.py]: Do we actually use the plain `Bunch` and `BunchConst` classes anywhere? Why do we need the tree of subclasses?

**Comment by wmvanvliet** [comment]
haven't read everything super carefully, but it looks like a nice UX improvement to me!

**Comment by larsoner** [comment]
And to answer explicitly the question that disappeared (?) in case someone else is curious, `Bunch` is kind of standard and used a few places, `BunchConst` is used a few places and has the property of not allowing rewriting values (useful for constants but not everywhere a `Bunch` is used), and `BunchConstNamed` is not suitable e.g. when values are `str` (it might be possible, but to be conservative it's safer to only support `int` and `float` for now)

**Comment by agramfort** [comment]
CIs fail for diff LGTM

**Comment by massich** [comment]
+1000
Cool I really like the bunch, and I'll move the digpoint to the refactoring.

**Pull request merged**